src/main/java/org/spongepowered/api/item/FireworkEffect.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item;
+
+import java.awt.Color;
+import java.util.List;
+
+/**
+ * Represents an explosion that a firework has.
+ *
+ * <p>{@link FireworkEffect}s are immutable once created. To change one
+ * or create one, use {@link FireworkEffectBuilder}.</p>
+ */
+public interface FireworkEffect {
+
+    /**
+     * Gets whether this {@link FireworkEffect} will flicker when
+     * detonated.
+     *
+     * @return Whether this effect will flicker
+     */
+    boolean flickers();
+
+    /**
+     * Gets whether this {@link FireworkEffect} will have a trail
+     * when detonated.
+     *
+     * @return Whether this effect will have a trail
+     */
+    boolean hasTrail();
+
+    /**
+     * Gets the ordered list of colors.
+     *
+     * <p>In some implementations, the order of colors defines the colors
```

Comment:
Again, it's all up to the implementation of the firework display client side. For all we care, the order may be ignored or the order may be reversed.